shared/actions/chat.js

```diff
@@ -1322,79 +1372,14 @@ function * _uploadAttachment ({param, conversationIDKey, outboxID}: {param: Chat
   const uploadDone = yield takeFromChannelMap(channelMap, 'chat.1.chatUi.chatAttachmentUploadDone')
   uploadDone.response.result()
 
-  const finished = yield join(finishedTask)
-  const {params: {messageID}} = finished
+  yield join(finishedTask)
   yield cancel(progressTask)
   yield cancel(previewTask)
   closeChannelMap(channelMap)
 
   return messageID
 }
 
-function * _selectAttachment ({payload: {conversationIDKey, filename, title, type, outboxID = (Math.ceil(Math.random() * 1e9) + '')}}: Constants.SelectAttachment): SagaGenerator<any, any> {
-  const clientHeader = yield call(_clientHeader, CommonMessageType.attachment, conversationIDKey)
-  const attachment = {
-    filename,
-  }
-
-  const username = yield select(usernameSelector)
-
-  yield put({
-    logTransformer: appendMessageActionTransformer,
-    payload: {
-      conversationIDKey,
-      messages: [_temporaryAttachmentMessageForUpload(
-        conversationIDKey,
-        username,
-        title,
-        filename,
-        outboxID,
-        type,
-      )],
-    },
-    type: 'chat:appendMessages',
-  })
-
-  const param = {
-    conversationID: keyToConversationID(conversationIDKey),
-    clientHeader,
-    attachment,
-    title,
-    metadata: null,
-    identifyBehavior: yield call(_getPostingIdentifyBehavior, conversationIDKey),
-  }
-
-  try {
-    const messageID = yield call(_uploadAttachment, {param, conversationIDKey, outboxID})
-    yield put(({
-      type: 'chat:updateTempMessage',
-      payload: {
-        conversationIDKey,
-        outboxID,
-        message: {type: 'Attachment', messageState: 'sent', messageID, key: messageID},
-      },
-    }: Constants.UpdateTempMessage))
-
-    yield put(({
-      type: 'chat:markSeenMessage',
```

Comment:
The `attachment` message now comes in through `incomingMessage` so it should be marked seen by the regular path.